website/content/docs/plugins/containerized-plugins.mdx

```diff
@@ -0,0 +1,253 @@
+
+    a. Default Docker install:
+        ```bash
+        sudo tee /etc/docker/daemon.json <<EOF
+        {
+        "runtimes": {
+          "runsc": {
+            "path": "/usr/local/bin/runsc",
+              "runtimeArgs": [
+                "--host-uds=all"
+              ]
+            }
+          }
+        }
+        EOF
+        sudo systemctl reload docker
+        ```
+    
+    b. Rootless Docker install:
+        ```bash
+        mkdir -p ~/.config/docker
+        tee ~/.config/docker/daemon.json  <<EOF
+        {
+          "runtimes": {
+            "runsc": {
+              "path": "/usr/local/bin/runsc",
+              "runtimeArgs": [
+                "--host-uds=all"
+                "--ignore-cgroups"
+              ]
+            }
+          }
+        }
+        EOF
+        systemctl --user restart docker
+        ```
+
+### Building plugin container images
+
+To be compatible with running in a container, plugins must be built with at
+least v1.5.0 of [go-plugin](https://github.com/hashicorp/go-plugin). This can
+be pulled in directly via `go get github.com/hashicorp/go-plugin@latest`, or
+through the SDK from at least v0.10.0: `go get github.com/hashicorp/vault/sdk@latest`.
+
+Plugin container images should run a binary the same way it would be expected
+to run as a plain plugin process. For most plugins this simply means running
+the binary as the entrypoint. For example, for the KV plugin:
+
+1. Build the binary
+    ```bash
+    git clone https://github.com/hashicorp/vault-plugin-secrets-kv.git
+    cd vault-plugin-secrets-kv/
+    GOOS=linux go build -o kv cmd/vault-plugin-secrets-kv/main.go
+    ```
+
+1. Create the following `Dockerfile`:
+    ```Dockerfile
+    FROM gcr.io/distroless/static-debian12
+
+    COPY kv /bin/kv
+
+    ENTRYPOINT [ "/bin/kv" ]
+    ```
+
+1. Build the image:
+    ```bash
+    docker build -t hashicorp/vault-plugin-secrets-kv .
+    ```
+
+The image entrypoint, command, and environment can be further configured at
+registration time by using the `-command`, `-args`, and `-env` flags for
+`vault plugin register`.
+
+### Registration
+
+Once Vault has a suitable container runtime available, the remaining steps are
+very similar to registering and using other external plugins. The config setting
+for [`plugin_directory`](/vault/docs/configuration#plugin_directory) is not required
+for containerized plugins, but any plugin containers must already be available
+locally when mounted.
+
+1. Store the SHA256 of the plugin image:
+    ```bash
+    SHA256="$(docker images --no-trunc --format="{{ .ID }}" hashicorp/vault-plugin-secrets-kv | cut -d: -f2)"
+    ```
+
+1. Register a plugin, specifying your container image as `-oci_image`:
+    ```bash
+    vault plugin register \
+        -sha256="$SHA256" \
+        -oci_image=hashicorp/vault-plugin-secrets-kv \
+      secret kv-container
+    ```
+
+1. Enable the plugin:
+    ```bash
+    vault secrets enable kv-container
+    ```
```

Comment:
````suggestion
    ```shell-session
    vault secrets enable CONTAINER_NAME
    ```
````

Reply:
@schavis I am curious about this pattern. It makes someone following these steps unable to simply copy and paste commands. They must also edit the `CONTAINER_NAME`. In my experience it is more preferable to not have to modify the command at all when I just want to get something running.

Reply:
Given that the kv plugin is just an example and not the only plugin folks may need to follow these steps for, the need to edit the line is true regardless. Using an obviously named placeholder makes the text more accessible to a broad range of readers because it makes it more obvious what information they need to provide and helps readers make connections between commands to see how the steps are related.

Reply:
@schavis Thanks for the reply! I can definitely see your point. I wonder if a good compromise would be to use an actual shell variable in the commands and set it at the beginning? Similar to what we do with SHA256? Something like

```bash
NAME=kv-container
TAG=hashicorp/vault-plugin-secrets-kv
$ # step 1 ...
$ vault plugin register \
        -sha256="$SHA256" \
        -oci_image="$TAG" \
      secret "$NAME"
$ vault secrets enable "$NAME"
```

That way anyone that wants to directly follow along can do so without any modifactions. And others can just modify a few shell variables.

Anyway, just a thought and I will of course differ to your expertise in this area 😄